src/liboslexec/rs_fallback.cpp

```diff
@@ -316,6 +317,18 @@ rs_trace_get(OSL::OpaqueExecContextPtr exec_ctx, OSL::ustringhash name,
 #endif
 }
 
+OSL_RSOP OSL_HOSTDEVICE void*
+rs_allocate_closure(OSL::OpaqueExecContextPtr exec_ctx, size_t size,
+                    size_t alignment)
+{
+#ifndef __CUDA_ARCH__
```

Comment:
As this is rs_fallback.cpp, it should never be compiled by CUDA, can we remove the #ifndef __CUDA_ARCH__

Reply:
Everything in rs_fallback is done this way. It is compiled by Cuda, is it not?

Reply:
Its is not, rs_fallback.cpp only exists in
`set (lib_src`
which is only compiled for the host, it turns around and just calls the virtual functions of renderer services (which are only on the host) and has no way to be customized for other targets.
More of a rs_legacy_adapter.cpp